File: patch_app.ts
import fs from 'fs';

const file = 'src/App.tsx';
let code = fs.readFileSync(file, 'utf8');

if (!code.includes('LineConfigurationView')) {
  code = code.replace(
    /import \{ PartMasterView \}/,
    "import { LineConfigurationView } from './views/LineConfigurationView';\nimport { PartMasterView }"
  );
  code = code.replace(
    /case 'part-master':/,
    "case 'line-configuration':\n        return <LineConfigurationView />;\n      case 'part-master':"
  );
  fs.writeFileSync(file, code);
  console.log('App.tsx patched');
}

File: patch_app.py
import sys

with open('src/App.tsx', 'r') as f:
    content = f.read()

import re

# Remove LineConfigurationView import
content = re.sub(r'import { LineConfigurationView } from \'./views/LineConfigurationView\';\n?', '', content)

# Replace 'unified-settings' with 'install' where initialTab="unified-settings"
content = content.replace('initialTab="unified-settings"', 'initialTab="install"')

# Replace 'spreadsheet' with 'install' where initialTab="spreadsheet"
content = content.replace('initialTab="spreadsheet"', 'initialTab="install"')

with open('src/App.tsx', 'w') as f:
    f.write(content)
